vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java

```diff
@@ -214,9 +218,80 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
         managerToTest.deleteTextBlockById(textBlockId, "slideId_1");
     }
 
+    @Test(expected = BlockDoesNotExistException.class)
+    public void test_deleteSpaceBlockById_forNonExistentId() throws BlockDoesNotExistException {
+        String spaceBlockId = "notARealId";
+        when(contentBlockRepository.findById("notARealId")).thenReturn(Optional.empty());
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(spaceBlockRepo).deleteById(spaceBlockId);
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(textBlockRepo).existsById(spaceBlockId);
+        managerToTest.deleteSpaceBlockById(spaceBlockId, "slideId_1");
+    }
+
+    @Test
+    public void test_deleteSpaceBlockById_success() throws BlockDoesNotExistException {
+        String spaceBlockId = "realId";
+        when(contentBlockRepository.findById(spaceBlockId)).thenReturn(Optional.of(contentBlock));
+        managerToTest.deleteSpaceBlockById(spaceBlockId, "slideId_1");
+        Mockito.verify(spaceBlockRepo).deleteById(spaceBlockId);
+
+    }
+
+    @Test
+    public void test_createSpaceBlock_success() {
+        String slideId = "slideId";
+        ISlide slide = null;
+        String title = "this is a space block";
+        Integer contentOrder = 2;
+        Space space = new Space();
+        ISpaceBlock spaceBlock = new SpaceBlock();
+        spaceBlock.setTitle(title);
+        spaceBlock.setId(slideId);
+        spaceBlock.setSpace(space);
+        Mockito.when(slideManager.getSlide(slideId)).thenReturn(slide);
+        Mockito.when(spaceBlockRepo.save((SpaceBlock) spaceBlock)).thenReturn((SpaceBlock) spaceBlock);
+        Mockito.when(spaceBlockFactory.createSpaceBlock(slide, title, (ISpace) space)).thenReturn(spaceBlock);
+        ISpaceBlock createdBlock = managerToTest.createSpaceBlock(slideId, title, contentOrder, space);
+        Assert.assertEquals(createdBlock.getContentOrder(), contentOrder);
+        Assert.assertEquals(createdBlock.getTitle(), title);
+        Assert.assertEquals(createdBlock.getId(), slideId);
+        Assert.assertEquals(createdBlock.getSpace(), space);
+    }
+
+    @Test
+    public void test_getSpaceBlock_success() {
+        String spaceId = "spaceId";
+        SpaceBlock spaceBlock = new SpaceBlock();
+        spaceBlock.setId(spaceId);
+        when(spaceBlockRepo.findById(spaceId)).thenReturn(Optional.of(spaceBlock));
+        ISpaceBlock retrievedSpaceBlock = managerToTest.getSpaceBlock(spaceId);
+        assertEquals(spaceId, retrievedSpaceBlock.getId());
+    }
+
+    @Test
+    public void test_getSpaceBlock_NonExistentId() throws BlockDoesNotExistException {
+        String spaceId = "notRealId";
+        when(spaceBlockRepo.findById(spaceId)).thenReturn(Optional.empty());
+        ISpaceBlock retrievedSpaceBlock = managerToTest.getSpaceBlock(spaceId);
+        assertNull(retrievedSpaceBlock);
+
+    }
+
+    @Test
+    public void test_updateSpaceBlock_success() {
```

Comment:
needs to follow our naming conventions

Reply:
this method is testing this function: managerToTest.saveSpaceBlock(spaceBlock); so do you mean it should be called test_saveSpaceBlock_success? if so... then there is already this function, but its saving a space block and this one is updating it, so what should I call this method?

Reply:
I'm sure this is a question you can figure out yourself... ;)